WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationSettingDetailsViewController.swift

```diff
@@ -0,0 +1,143 @@
+import Foundation
+
+
+/**
+*  @class           NotificationSettingDetailsViewController
+*  @brief           The purpose of this class is to render a collection of NotificationSettings for a given
+*                   Stream, encapsulated in the class NotificationSettings.Stream, and to provide the user
+*                   a simple interface to update those settings, as needed.
+*/
+
+public class NotificationSettingDetailsViewController : UITableViewController
+{
+    // MARK: - View Lifecycle
+    public override func viewDidLoad() {
+        super.viewDidLoad()
+        setupTableView()
+    }
+
+    public override func viewWillDisappear(animated: Bool) {
+        super.viewWillDisappear(animated)
+        saveSettingsIfNeeded()
+    }
+    
+    
+    
+    // MARK: - Setup Helpers
+    private func setupTableView() {
+        // Register the cells
+        tableView.registerClass(SwitchTableViewCell.self, forCellReuseIdentifier: reuseIdentifier)
+                
+        // Hide the separators, whenever the table is empty
+        tableView.tableFooterView = UIView()
+        
+        // Style!
+        WPStyleGuide.configureColorsForView(view, andTableView: tableView)
+    }
+    
+    
+    
+    // MARK: - Public Helpers
+    public func setupWithSettings(settings: NotificationSettings, stream: NotificationSettings.Stream) {
+        // Title
+        switch settings.channel {
+        case .WordPressCom:
+            title = NSLocalizedString("WordPress.com Updates", comment: "WordPress.com Notification Settings Title")
+        default:
+            title = stream.kind.description()
+        }
+        
+        // Structures
+        self.settings   = settings
+        self.stream     = stream
+        self.newValues  = [String: Bool]()
+        
+        tableView.reloadData()
+    }
+    
+    
+    
+    // MARK: - UITableView Delegate Methods
+    public override func numberOfSectionsInTableView(tableView: UITableView) -> Int {
+        return sectionCount
+    }
+    
+    public override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return settings?.sortedPreferenceKeys(stream).count ?? emptyRowCount
+    }
+    
+    public override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCellWithIdentifier(reuseIdentifier) as! SwitchTableViewCell
+        
+        configureCell(cell, indexPath: indexPath)
+        
+        return cell
+    }
+    
+    
+    
+    // MARK: - UITableView Helpers
+    private func configureCell(cell: SwitchTableViewCell, indexPath: NSIndexPath) {
+        let preferences         = stream?.preferences
+        let sortedKeys          = settings?.sortedPreferenceKeys(stream)
+        let key                 = sortedKeys?[indexPath.row]
```

Comment:
Just thinking a bit about the unboxing here.  If for some reason `sortedKeys` was `nil`, the assignment to `key` would also be `nil`. Then when `key` is used in the assignment to `cell.name` or `cell.isOn` ... kablooey?  At least this is what I gather from this playground snippet:
```
var foo : String? = nil
let bar = foo
let baz = bar!.isEmpty
```